test/errors.tsx

```diff
@@ -40,9 +40,11 @@ test('catch and display error', t => {
 			" 22:     throw new Error('Oh no');",
 			' 23:   };',
 			' 24:',
-			' 25:   render(<Test />, {stdout});',
+			' 25:   const {waitUntilExit} = render(<Test />, {stdout});',
 			'',
 			' - Test (test/errors.tsx:22:9)'
 		]
 	);
+
+	await t.throwsAsync(waitUntilExit, {message: 'Oh no'});
```

Comment:
fix for `Unhandled rejection in test/errors.tsx`

Reply:
> To avoid it, we should add something like a isWaitingForExit boolean property to Ink and set it to true in waitUntilExit method.

With synchronous errors `waitUntilExit` is never called. I can add some prop to render method maybe. Wdyt?

Reply:
Hm, wasn't that the purpose of this PR? To resolve/reject the promise returned by `waitUntilExit` for synchronous errors?

Reply:
Hm, that's not good. Whatever solution we decide to go with, it needs to work with this use case, but without unhandled rejection promises.

Reply:
Hey @vadimdemedes. I've pushed new implementation, please have a look 🙂

Reply:
I'm not a fan of an option that makes `waitUntilExit` work. As a user of Ink, I'd expect it to just work when I call it.

It looks like this problem isn't solvable without breaking changes. Perhaps we can think what the better API would look like if we weren't constrained by the presence of `waitUntilExit` and ship this in a major release.